lib/dealership.rb

```diff
@@ -0,0 +1,48 @@
+require './lib/car'
+
+class Dealership
+  attr_reader :name, :addy, :inventory
+  def initialize(name, addy)
+    @name = name
+    @addy = addy
+    @inventory = []
+  end
+
+  def inventory_count
+    return @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    if @inventory == []
+      return false
+    else
+      return true
+    end
+  end
+
+  def cars_by_make(type)
+    return @inventory.make(type)
+  end
+
+  def total_value
+    @total = []
+    @inventory.each do |value|
+    @total << value.total_cost
```

Comment:
You were very close here; the `<<` (shovel) operator adds to an array. But we're not expecting an array from the `#total_value` method, we're expecting an Integer. So what gets returned is `[54000]`. The interaction pattern expects an Integer, as so: 
```
pry(main)> dealership.total_value
#=> 156000
```


You were correct in doing a `.each` through the inventory, but all we need is something like: 
`@total += value.total_cost`
and then return that, which you do later. Also, `@total` should not be an array on line 32, it can just be initialized to 0. Finally, it does not need to be a class-level variable with `@`, since it just returns a value from this one method and is not accessed anywhere else in this class file.